scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,82 @@
+
+For its purpose, this approach needs to be revised.
+
+1. The vulnerability schema rarely changes, but maintaining multiple bundles is expensive.  It requires multiple GitHub workflow runs, which increases the chances of failure and pressure on the security sources' endpoints.
+2. It isn't easy to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process is noisy, convoluted, and complex.
+3. The offline mode's "latest bundle" needs to exist as a complex solution to help users transition to the versioned bundle.  It contains all versions in a single bundle.  But one bundle per release makes the latest bundle large and, most of the time, essentially carries the same data since the schema is the same.
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
+
+Initially, two version streams will exist (table 1 below):
+
+| Stream | Git Ref      | Description                                                        |
+|--------|--------------|--------------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema                 |
+| v1     | heads/4.4.5  | This stream tracks the schema used by the current Stackrox release |
+
+On the subsequent releases (for example,. `4.4.6`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release (`heads/4.4.6`).
+2.  The schema has changed, so v1 stays at current release (`heads/4.4.5`), and a new version, v2, is created to track the next release (`heads/4.4.6`).
+
+### Building bundle version in Exporter
+
+The GitHub workflow will have a map with "bundle version" as keys, and "git reference" as values, similar to table 1 above.  This can be stored statically in the matrix definition.
+
+For each version in the mapping, a bundle will be built based on the specified git ref and stored at `https://definitions.stackrox.io/v4/<bundle-version>/vulnerabilities.zip`.
+
+In every release cut, the key specified in `scanner/VULNERABILITY_VERSION` will be updated in the mapping to the current release branch reference (e.g., `heads/4.4.6`).  Notice that if the version is the same, it will be updated.  If it's new, it will be added, and the previous version will stay.
+
+### Selection bundle version in Scanner
+
+Scanner will determine which version bundle it wants based on hard-coded information instead during build time.  A file at `scanner/VULNERABILITY_VERSION` will determine the version on release builds and nightlies.  On non-release builds this will be set to "dev".
+
+Today scanner supports a special variable `ROX_VERSION` in the vulnerability URL configuration.  A new variable will be added called `ROX_SCANNER_VULNERABILITY_VERSION` to specify the expected bundle version to fetch.
```

Comment:
I think staging (nightly build) can stop relying on "dev" version since the ADR mentioned
> A new variable will be added called `ROX_SCANNER_VULNERABILITY_VERSION` to specify the expected bundle version to fetch.

When a nightly is built, **_ROX_SCANNER_VULNERABILITY_VERSION_** should be pointing to the associate bundle version. Is this understanding correct?

Reply:
That's correct. It means if we bump the schema version in master, a new version will be provided in the `ROX_SCANNER_VULNERABILITY_VERSION`. After that, the workflow will be "dev" with the new schema, but since nightlies are using the version, it will only consume the new schema once it gets deployed.